citystatmodel.cpp

```diff
@@ -0,0 +1,207 @@
+        else if (index.column() == 1)
+            returnValue = item.countryName;
+        else if (index.column() == 2)
+            returnValue = item.cityName;
+        else if (index.column() == 3)
+            returnValue = item.cityPopulation;
+        else if (index.column() == 4)
+            returnValue = item.recordHighTempereture;
+        else if (index.column() == 5)
+            returnValue = item.recordLowTempereture;
+     }
+     return returnValue;
+}
+
+bool CityStatModel::setData(const QModelIndex &index, const QVariant &value, int role)
+{
+    if (data(index, role) != value)
+    {
+        switch (index.column())
+        {
+        case 0:
+        {
+            m_cityItemsList[index.row()].continentName=value.toString();
+            break;
+        }
+        case 1:
+        {
+            m_cityItemsList[index.row()].countryName=value.toString();
+            break;
+        }
+        case 2:
+        {
+            m_cityItemsList[index.row()].cityName=value.toString();
+            break;
+        }
+        case 3:
+        {
+            m_cityItemsList[index.row()].cityPopulation=value.toUInt();
+            fillPopulationInfo();
+            break;
+        }
+        case 4:
+        {
+            m_cityItemsList[index.row()].recordHighTempereture=value.toInt();
+            break;
+        }
+        case 5:
+        {
+            m_cityItemsList[index.row()].recordLowTempereture=value.toInt();
+            break;
+        }
+        }
+        emit dataChanged(index, index, QVector<int>() << role);
+        return true;
+    }
+    return false;
+}
+
+Qt::ItemFlags CityStatModel::flags(const QModelIndex &index) const
+{
+    return QAbstractTableModel::flags(index) | Qt::ItemIsEditable;
+}
+
+bool CityStatModel::insertRows(int row, int count, const QModelIndex &parent)
+{
+    beginInsertRows(parent, row, row + count - 1);
+    for (int i = 0; i<count;i++)
+    {
+        CityItem item;
+        item.continentName="";
+        item.countryName="";
+        item.cityName="";
+        item.cityPopulation = 0;
+        item.recordHighTempereture = 0;
+        item.recordLowTempereture = 0;
+        m_cityItemsList.push_back(item);
+    }
+    endInsertRows();
+    return true;
+}
+
+bool CityStatModel::removeRows(int row, int count, const QModelIndex &parent)
+{
+    beginRemoveRows(parent, row, row + count - 1);
+    // FIXME: Implement me!
+    endRemoveRows();
+}
+
+void CityStatModel::saveDataInitiated()
+{
+    unsigned int savedItemsCounter = 0;
+    bool isSavingSuccessfull = m_dataLoader->writeCityItemsList(m_cityItemsList, savedItemsCounter);
+    emit isSaveDataSuccessfull(isSavingSuccessfull, savedItemsCounter);
+}
+
+void CityStatModel::fillPopulationInfo()
+{
+    m_minPopulationElement = std::numeric_limits<unsigned int>::max();
+    m_maxPopulationElement = 0;
+    m_totalPopulationElement = 0;
+
+    for(int i = 0; i < m_cityItemsList.size(); i++)
```

Comment:
Replace this for loop with modern range-based: for (auto &cityItem : m_cityItemsList)

Reply:
done